webapp/src/components/jira_field.jsx

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ReactSelectSetting from 'components/react_select_setting';
+import Input from 'components/input';
+
+export default class JiraField extends React.PureComponent {
+    static propTypes = {
+        id: PropTypes.object.isRequired,
+        field: PropTypes.object.isRequired,
+        obeyRequired: PropTypes.bool,
+        onChange: PropTypes.func,
+        value: PropTypes.any,
+    };
+
+    static defaultProps = {
+        obeyRequired: true,
+    };
+
+    handleChange = (id, value) => {
+        if (this.props.onChange) {
+            this.props.onChange(id, value);
+        }
+    };
+
+    allowedValueToSelectValue = (allowedValue) => {
```

Comment:
Maybe a comment? Took me a couple minutes to figure out what this function did. Also, maybe slightly different name, `makeSelectValueFromAllowedValue`, or maybe even just `makeSelectValue = (allowedValue)...` would be more intention-revealing?